src/agent/src/image.rs

```diff
@@ -0,0 +1,377 @@
+        if !cc_pause_bundle.exists() {
+            return Err(anyhow!("Pause image not present in rootfs"));
+        }
+
+        info!(sl(), "use guest pause image cid {:?}", cid);
+        let pause_bundle = Path::new(CONTAINER_BASE).join(cid).join(target_subpath);
+        let pause_rootfs = pause_bundle.join("rootfs");
+        fs::create_dir_all(&pause_rootfs)?;
+
+        let copy_if_not_exists = |src: &Path, dst: &Path| -> Result<()> {
+            if !dst.exists() {
+                fs::copy(src, dst)?;
+            }
+            Ok(())
+        };
+        copy_if_not_exists(
+            &cc_pause_bundle.join(CONFIG_JSON),
+            &pause_bundle.join(CONFIG_JSON),
+        )?;
+        copy_if_not_exists(
+            &cc_pause_bundle.join("rootfs/pause"),
+            &pause_rootfs.join("pause"),
+        )?;
+
+        Ok(pause_rootfs.display().to_string())
+    }
+
+    /// pull_image is used for call image-rs to pull image in the guest.
+    /// # Parameters
+    /// - `image`: Image name (exp: quay.io/prometheus/busybox:latest)
+    /// - `cid`: Container id
+    /// - `image_metadata`: Annotations about the image (exp: "containerd.io/snapshot/cri.layer-digest": "sha256:24fb2886d6f6c5d16481dd7608b47e78a8e92a13d6e64d87d57cb16d5f766d63")
+    /// # Returns
+    /// - The image rootfs bundle path. (exp. /run/kata-containers/cb0b47276ea66ee9f44cc53afa94d7980b57a52c3f306f68cb034e58d9fbd3c6/images/rootfs)
+    pub async fn pull_image(
+        &self,
+        image: &str,
+        cid: &str,
+        image_metadata: &HashMap<String, String>,
+    ) -> Result<String> {
+        info!(sl(), "image metadata: {:?}", image_metadata);
+        Self::set_proxy_env_vars();
+
+        let is_sandbox = if let Some(value) = image_metadata.get("io.kubernetes.cri.container-type")
+        {
+            value == "sandbox"
+        } else if let Some(value) = image_metadata.get("io.kubernetes.cri-o.ContainerType") {
+            value == "sandbox"
+        } else {
+            false
+        };
+
+        if is_sandbox {
+            let mount_path = Self::unpack_pause_image(cid, "pause")?;
+            self.add_image(String::from(image), String::from(cid)).await;
+            return Ok(mount_path);
+        }
+
+        // Image layers will store at KATA_IMAGE_WORK_DIR, generated bundles
+        // with rootfs and config.json will store under CONTAINER_BASE/cid/images.
+        let bundle_path = Path::new(CONTAINER_BASE).join(cid).join("images");
+        fs::create_dir_all(&bundle_path)?;
+        info!(sl(), "pull image {:?}, bundle path {:?}", cid, bundle_path);
+
+        let res = self
+            .image_client
+            .lock()
+            .await
+            .pull_image(image, &bundle_path, &None, &None)
+            .await;
+        match res {
+            Ok(image) => {
+                info!(
+                    sl(),
+                    "pull and unpack image {:?}, cid: {:?}, with image-rs succeed. ", image, cid
```

Comment:
You can simplify this as follows (and I'd remove the "with image-rs" as I don't think that's relevant (any more)?

```suggestion
                    "pull and unpack image {image:?}, cid: {cid:?} succeeded"
```

Same comment for other instances.

Reply:
Fixed.